Lib/typing.py

```diff
@@ -516,6 +521,22 @@ def TypeAlias(self, parameters):
     raise TypeError(f"{self} is not subscriptable")
 
 
+@_SpecialForm
+def Concatenate(self, parameters):
+    """Used in conjunction with ParamSpec and Callable to represent a higher
+    order function which add, removes or transform parameters of a Callable.
+    """
+    if parameters == ():
+        raise TypeError("Cannot take a Concatenate of no types.")
+    if not isinstance(parameters, tuple):
+        parameters = (parameters,)
+    msg = "Concatenate[arg, ...]: each arg must be a type."
+    parameters = tuple(_type_check(p, msg) for p in parameters)
+    if not any(isinstance(p, ParamSpec) for p in parameters):
+        raise TypeError("Concatenate must contain at least one ParamSpec variable.")
```

Comment:
According to PEP 612, the final parameter must be a `ParamSpec` (I can't tell whether other parameters may also be `ParamSpec`.)